packages/swap/prisma/schema.prisma

```diff
@@ -266,3 +283,16 @@ model ChainTracking {
 
   @@schema("public")
 }
+
+model StateChainError {
+  id              Int             @id @default(autoincrement())
+  specVersion     Int
+  palletIndex     Int
+  errorIndex      Int
+  name            String
+  docs            String
+  ignoredEgresses IgnoredEgress[] @ignore
+
+  @@unique([specVersion, palletIndex, errorIndex])
+  @@schema("public")
+}
```

Comment:
oof.. is this the pattern we're following  moving forward?

Reply:
i thought its a bit convoluted at first but after thinking it through i think its quite good if we can do this for different kind of errors

Reply:
maybe we could connect the `StateChainError` to the swap directly (without the `IgnoredEgress`)?

Reply:
> maybe we could connect the `StateChainError` to the swap directly (without the `IgnoredEgress`)?

Unfortunately not, the egress models contain the egress amount.

> oof.. is this the pattern we're following moving forward?

Yeah sadly because the state chain errors need to be looked up via rpc

Reply:
> Unfortunately not, the egress models contain the egress amount.

i think you are not using the `amount` of the `IgnoredEgress`, right?

Reply:
The `EgressIgnored` event has this error:
```json
{
  "asset": {
    "__kind": "Btc"
  },
  "amount": "11",
  "reason": {
    "value": {
      "error": "0x06000000",
      "index": 34
    },
    "__kind": "Module"
  },
  "swapId": "2"
}
```

We have to convert the `reason` to a human-readable error message which involves looking it up in the runtime metadata. For historic swaps, we have to ensure that we are using the runtime metadata for the block when the `EgressIgnored` event happens. The easiest way to do this is imo is to look up the error message in the metadata and save it here.

I'm open to other implementations/approaches

Reply:
> > Unfortunately not, the egress models contain the egress amount.
> 
> i think you are not using the `amount` of the `IgnoredEgress`, right?

You're right, I should be

Reply:
No I actually am, `packages/swap/src/event-handlers/swapEgressIgnored.ts`, line 96

Reply:
> No I actually am, `packages/swap/src/event-handlers/swapEgressIgnored.ts`, line 96

yeah you are storing it, but we dont expose it anywhere afterwards i think 🙂 so for the information we expose right now, we would not need the `IgnoredEgress` model, connecting the swap directly to a `StateChainError` would be enough

Reply:
so the swap could have an `error` property that we could maybe use for different errors (not only the ones caused by `IgnoredEgress`)

Reply:
I've added the `ignoredEgress` information to the response so we can generate a block explorer link, etc

Reply:
> so the swap could have an error property that we could maybe use for different errors (not only the ones caused by IgnoredEgress)

this is how it works, it takes a generic `name` and `message`